Build failing when no repo tags

As explained, the break was due to a lack of wifi at my aged-care facility.
A friend managed to set me up with wifi via my phone.

cd c:\z390dev\z390

entered ..... build.bat
got
c:\z390dev\z390>build.bat
Could Not Find c:\z390dev\z390\jar\z390.jar
src\az390.java
src\ez390.java
src\gz390.java
src\gz390_screen.java
src\iz390.java
src\lz390.java
src\MessageBox.java
src\mz390.java
src\pz390.java
src\sz390.java
src\test_screen.java
src\test_unicode.java
src\tz390.java
src\vz390.java
src\z390.java
src\zACB.java
src\zc390.java
src\zCatVAIX.java
src\zCatVCLR.java
       19 file(s) copied.
        1 file(s) copied.
java version "17.0.1" 2021-10-19 LTS
Java(TM) SE Runtime Environment (build 17.0.1+12-LTS-39)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.1+12-LTS-39, mixed mode, sharing)
gz390.java:574: warning: [removal] SecurityManager in java.lang has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
            ^
gz390.java:574: warning: [removal] getSecurityManager() in System has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
                                       ^
z390.java:423: warning: [removal] SecurityManager in java.lang has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
            ^
z390.java:423: warning: [removal] getSecurityManager() in System has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
                                       ^
4 warnings
fatal: No names found, cannot describe anything.
Version is uncontrolled - add FORCE parm to ignore
bat\bldjar ERROR: Error finding z390 version number
build.bat ERROR: Encountered RC 8 - exiting

c:\z390dev\z390>

John Hennesy

I used the FORCE parameter and got

c:\z390dev\z390>build.bat force
Could Not Find c:\z390dev\z390\jar\z390.jar
src\az390.java
src\ez390.java
src\gz390.java
src\gz390_screen.java
src\iz390.java
src\lz390.java
src\MessageBox.java
src\mz390.java
src\pz390.java
src\sz390.java
src\test_screen.java
src\test_unicode.java
src\tz390.java
src\vz390.java
src\z390.java
src\zACB.java
src\zc390.java
src\zCatVAIX.java
src\zCatVCLR.java
       19 file(s) copied.
        1 file(s) copied.
java version "17.0.1" 2021-10-19 LTS
Java(TM) SE Runtime Environment (build 17.0.1+12-LTS-39)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.1+12-LTS-39, mixed mode, sharing)
gz390.java:574: warning: [removal] SecurityManager in java.lang has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
            ^
gz390.java:574: warning: [removal] getSecurityManager() in System has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
                                       ^
z390.java:423: warning: [removal] SecurityManager in java.lang has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
            ^
z390.java:423: warning: [removal] getSecurityManager() in System has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
                                       ^
4 warnings
fatal: No names found, cannot describe anything.
Version is uncontrolled - add FORCE parm to ignore
bat\bldjar ERROR: Error finding z390 version number
build.bat ERROR: Encountered RC 8 - exiting

Abe,
I followed your instructions

1.  GO TO GITHUB.COM TO YOUR OWN FORK.
2.  ON THE PULL-DOWN FOR BRANCHES,
    SELECT THE MAIN BRANCH.
    I entere "github.com/chookperson/z390".
3.  JUST BELOW THE GREEN "CODE" BUTTON, YOU WILL FIND A
    "SYNC FORK" BUTTON, USE THAT TO UPDATE YOUR FORK.
    It says "this branch is out of date" but has a green
    Update button below. Pressed that.
    It replied
    "Successfully fetched and fast-forwarded from upstream 
     z390development:main."

4.  ON YOUR LOCAL MACHINE GO TO THE DIRECTORY THAT HOLDS YOUR
    CLONE OF THE REPOSITORY.
    cd c:\z390dev\z390

5.  ISSUE GIT STATUS - TO VALIDATE YOU'RE OK.
    It replies
    "fatal: cannot handle STATUS as a builtin"
    must be in lower-case
    "On branch DB2
     nothing to commit, working tree clean"

6.  ISSUE GIT SWITCH MAIN - TO VALIDATE YOU'RE OK.
    git switch main
    It replies
    "Switched to branch 'main'
     Your branch is up to date with 'origin/main'."

7.  ISSUE GIT STATUS - TO VALIDATE YOU'RE ON THE MAIN BRANCH NOW
    git status
    It replies
    "On branch main
     Your branch is up to date with 'origin/main'.
     nothing to commit, working tree clean"

8.  ISSUE GIT PULL - TO UPDATE YOUR MAIN BRANCH
    git pull
    It replies
    "remote: Enumerating objects: 70, done.
     remote: Counting objects: 100% (54/54), done.
     remote: Compressing objects: 100% (40/40), done.
     remote: Total 70 (delta 7), reused 31 (delta 4), pack-reused 16
     Unpacking objects: 100% (70/70), 110.26 KiB | 149.00 KiB/s, done.
     From https://github.com/chookperson/z390
        0f9cdbd..b68df64  main       -> origin/main
     Updating 0f9cdbd..b68df64
     Fast-forward
      .github/workflows/build.yml                        |   2 +-
      .github/workflows/docs.yml                         |   2 +-
      BUILD.BAT                                          |  18 +-
      build.sh                                           |  18 +-
      doc/contributing/contribute_code.md                |   2 +-
      doc/contributing/contributing_zcobol.md            |  51 ++
      doc/getting_started/quickstart.md                  |   5 +-
      doc/getting_started/zCOBOL_Overview.md             | 159 +++++
      doc/reference/.pages                               |   1 -
      doc/reference/assist.md                            |   4 +-
      doc/reference/structured_program.md                |   2 +-
      .../zCOBOL_NIST_COBOL_1985_Test_Results.md         | 717 +++++++++++++++++++++
      doc/reference/zCOBOL_System_Programmer_Guide.md    | 268 ++++++++
      doc/user_guide/.pages                              |   4 +
      doc/user_guide/{ => z390}/commands.md              |   6 +-
      doc/user_guide/{ => z390}/structured_macro.md      |   0
      doc/user_guide/{options => z390}/z390_filetypes.md |   0
      doc/user_guide/{options => z390}/z390_options.md   |   0
      doc/{reference => user_guide}/zCICS/index.md       |   0
      .../zCICS/zcics_app_prog_guide.md                  |   0
      doc/user_guide/zCOBOL/.pages                       |   5 +
      doc/user_guide/zCOBOL/commands.md                  |  37 ++
      .../zCOBOL/zCOBOL_System_Programmer_Guide.md       | 282 ++++++++
      doc/user_guide/zCOBOL/zCOBOL_User_Guide.md         | 164 +++++
      .../{options => zCOBOL}/zCOBOL_filetypes.md        |   0
      .../{options => zCOBOL}/zCOBOL_options.md          |   0
      doc/user_guide/zCOBOL_User_Guide.md                | 202 ++++++
      z390test/.gitignore                                |  48 ++
      z390test/LICENSE                                   | 339 ++++++++++
      z390test/README.md                                 | 242 +++++++
      z390test/build.gradle                              |  22 +
      z390test/gradle/wrapper/gradle-wrapper.jar         | Bin 0 -> 59821 bytes
      z390test/gradle/wrapper/gradle-wrapper.properties  |   5 +
      z390test/gradlew                                   | 234 +++++++
      z390test/gradlew.bat                               |  89 +++
      z390test/settings.gradle                           |   2 +
      .../src/main/groovy/org/z390/test/z390Test.groovy  | 274 ++++++++
      .../test/groovy/org/z390/test/RunAsmDemos.groovy   |  40 ++
      .../test/groovy/org/z390/test/RunAsmTests.groovy   | 101 +++
      .../src/test/groovy/org/z390/test/RunAssist.groovy |  48 ++
      .../test/groovy/org/z390/test/RunCblDemos.groovy   |  63 ++
      .../test/groovy/org/z390/test/RunCblTests.groovy   |  29 +
      .../test/groovy/org/z390/test/RunCmdProc.groovy    |  89 +++
      .../src/test/groovy/org/z390/test/RunRTBSAM.groovy |  38 ++
      .../src/test/groovy/org/z390/test/RunRTQSAM.groovy |  34 +
      .../test/groovy/org/z390/test/RunRtTests.groovy    | 104 +++
      .../test/groovy/org/z390/test/RunSortTests.groovy  |  76 +++
      .../test/groovy/org/z390/test/RunTestOption.groovy | 109 ++++
      .../groovy/org/z390/test/RunZSTRMacTest.groovy     | 165 +++++
      .../test/groovy/org/z390/test/RunZparTests.groovy  |  26 +
      .../test/groovy/org/z390/test/TestZ390Test.groovy  |  62 ++
      51 files changed, 4149 insertions(+), 39 deletions(-)
      create mode 100644 doc/contributing/contributing_zcobol.md
      create mode 100644 doc/getting_started/zCOBOL_Overview.md
      create mode 100644 doc/reference/zCOBOL_NIST_COBOL_1985_Test_Results.md
      create mode 100644 doc/reference/zCOBOL_System_Programmer_Guide.md
      create mode 100644 doc/user_guide/.pages
      rename doc/user_guide/{ => z390}/commands.md (95%)
      rename doc/user_guide/{ => z390}/structured_macro.md (100%)
      rename doc/user_guide/{options => z390}/z390_filetypes.md (100%)
      rename doc/user_guide/{options => z390}/z390_options.md (100%)
      rename doc/{reference => user_guide}/zCICS/index.md (100%)
      rename doc/{reference => user_guide}/zCICS/zcics_app_prog_guide.md (100%)
      create mode 100644 doc/user_guide/zCOBOL/.pages
      create mode 100644 doc/user_guide/zCOBOL/commands.md
      create mode 100644 doc/user_guide/zCOBOL/zCOBOL_System_Programmer_Guide.md
      create mode 100644 doc/user_guide/zCOBOL/zCOBOL_User_Guide.md
      rename doc/user_guide/{options => zCOBOL}/zCOBOL_filetypes.md (100%)
      rename doc/user_guide/{options => zCOBOL}/zCOBOL_options.md (100%)
      create mode 100644 doc/user_guide/zCOBOL_User_Guide.md
      create mode 100644 z390test/.gitignore
      create mode 100644 z390test/LICENSE
      create mode 100644 z390test/README.md
      create mode 100644 z390test/build.gradle
      create mode 100644 z390test/gradle/wrapper/gradle-wrapper.jar
      create mode 100644 z390test/gradle/wrapper/gradle-wrapper.properties
      create mode 100755 z390test/gradlew
      create mode 100644 z390test/gradlew.bat
      create mode 100644 z390test/settings.gradle
      create mode 100644 z390test/src/main/groovy/org/z390/test/z390Test.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunAsmDemos.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunAsmTests.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunAssist.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunCblDemos.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunCblTests.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunCmdProc.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunRTBSAM.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunRTQSAM.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunRtTests.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunSortTests.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunTestOption.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunZSTRMacTest.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/RunZparTests.groovy
      create mode 100644 z390test/src/test/groovy/org/z390/test/TestZ390Test.groovy"

9.  ISSUE BUILD.BAT - TO SEE IF YOU CAN NOW CREATE THE Z390
    EXECUTABLES.
    build.bat
    It replies
    "Could Not Find C:\z390dev\z390\jar\z390.jar
     src\az390.java
     src\ez390.java
     src\gz390.java
     src\gz390_screen.java
     src\iz390.java
     src\lz390.java
     src\MessageBox.java
     src\mz390.java
     src\pz390.java
     src\sz390.java
     src\test_screen.java
     src\test_unicode.java
     src\tz390.java
     src\vz390.java
     src\z390.java
     src\zACB.java
     src\zc390.java
     src\zCatVAIX.java
     src\zCatVCLR.java
             19 file(s) copied.
             1 file(s) copied.
     java version "17.0.1" 2021-10-19 LTS
     Java(TM) SE Runtime Environment (build 17.0.1+12-LTS-39)
     Java HotSpot(TM) 64-Bit Server VM (build 17.0.1+12-LTS-39, mixed mode, sharing)
     gz390.java:574: warning: [removal] SecurityManager in java.lang has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
            ^
     gz390.java:574: warning: [removal] getSecurityManager() in System has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
                                       ^
     z390.java:423: warning: [removal] SecurityManager in java.lang has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
            ^
     z390.java:423: warning: [removal] getSecurityManager() in System has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
                                       ^
     4 warnings
     fatal: No names found, cannot describe anything.
     Version is uncontrolled - add FORCE parm to ignore
     bat\bldjar ERROR: Error finding z390 version number
     build.bat ERROR: Encountered RC 8 - exiting"
   
Should I rerun with force?

Guys,
I'd like to thank you both for your patience and endurance.

We're almost there ....

c:\z390dev\z390>build.bat
Could Not Find c:\z390dev\z390\jar\z390.jar
src\az390.java
src\ez390.java
src\gz390.java
src\gz390_screen.java
src\iz390.java
src\lz390.java
src\MessageBox.java
src\mz390.java
src\pz390.java
src\sz390.java
src\test_screen.java
src\test_unicode.java
src\tz390.java
src\vz390.java
src\z390.java
src\zACB.java
src\zc390.java
src\zCatVAIX.java
src\zCatVCLR.java
       19 file(s) copied.
        1 file(s) copied.
java version "17.0.1" 2021-10-19 LTS
Java(TM) SE Runtime Environment (build 17.0.1+12-LTS-39)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.1+12-LTS-39, mixed mode, sharing)
gz390.java:574: warning: [removal] SecurityManager in java.lang has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
            ^
gz390.java:574: warning: [removal] getSecurityManager() in System has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
                                       ^
z390.java:423: warning: [removal] SecurityManager in java.lang has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
            ^
z390.java:423: warning: [removal] getSecurityManager() in System has been deprecated and marked for removal
            SecurityManager sm = System.getSecurityManager();
                                       ^
4 warnings
z390 version "v1.8.1-7-gb68df64"
'jar.exe' is not recognized as an internal or external command,
operable program or batch file.
bat\bldjar ERROR: Error creating z390.jar file
build.bat ERROR: Encountered RC 9009 - exiting

c:\z390dev\z390>

John

--- a/bash/relver
+++ b/bash/relver
@@ -9,6 +9,13 @@ cd ..
 
 VER="Uncontrolled";
 
+get_tags() {
+    if ! git config remote.upstream.url > /dev/null; then
+        git remote add upstream https://github.com/z390development/z390.git
+    fi
+    git fetch -p upstream
+}
+
 # is git installed
 $(git --version > /dev/null 2>&1)
 GIT_INSTALLED=$?
@@ -21,6 +28,12 @@ else
     if [ ${GIT_REPO} != 0 ]; then
         echo >&2 "Not a git repo.";
     else
+        $(git describe > /dev/null 2>&1)
+        GIT_DESCRIBE=$?
+        if [ ${GIT_DESCRIBE} != 0 ]; then
+            echo >&2 "Unable to git describe - fetching tags from upstream"
+            get_tags
+        fi
         VER=$(git describe);
     fi    
 fi

--- a/bat/RELVER.BAT
+++ b/bat/RELVER.BAT
@@ -13,16 +13,49 @@ rem ----- Lvl(%z_NestLevel%) Start %0 %1 %2 %3 %4 %5 %6 %7 %8 %9
 pushd %~dps0..
 set z_Version="Uncontrolled"
 
-git --man-path >NUL
-
+rem -- is git installed?
+git --version >NUL
 set z_Git_Installed=%ERRORLEVEL%
 if %z_Git_Installed% NEQ 0 (echo Git is not installed
                             goto final
-                            ) 
+                            )
 
+rem -- are we in a git repository?
+git rev-parse --is-inside-work-tree >NUL
+set z_Git_WorkTree=%ERRORLEVEL%
+if %z_Git_WorkTree% NEQ 0 (echo Git cannot determine whether we're in a work tree
+                           goto final
+                           )
 for /F "tokens=* USEBACKQ" %%F in (`git rev-parse --is-inside-work-tree`) do (set z_Git_Repo=%%F)
 if "%z_Git_Repo%" NEQ "true" (echo Not a git repository
                               goto final)
+
+rem -- get git to describe the repo
+git describe >NUL
+set z_Git_Describe=%ERRORLEVEL%
+if %z_Git_Describe% EQU 0 goto describe
+echo Unable to git describe - fetching tags from upstream
+
+rem -- make sure we have an upstream defined
+git config remote.upstream.url >NUL
+set z_Git_Remote=%ERRORLEVEL%
+if %z_Git_Remote% EQU 0 goto fetch_upstream
+rem -- use default to set the remote location
+git remote add upstream https://github.com/z390development/z390.git
+set z_Git_SetRemote=%ERRORLEVEL%
+if %z_Git_SetRemote% NEQ 0 (echo Git cannot set remote
+                            goto final
+                            )
+
+:fetch_upstream
+rem -- fetch upstream to fix missing git tags
+git fetch -p upstream
+set z_Git_Fetch=%ERRORLEVEL%
+if %z_Git_Fetch% NEQ 0 (echo Git cannot fetch from upstream
+                        goto final
+                        )
+
+:describe
 for /F "tokens=* USEBACKQ" %%I in (`git describe`) do set z_Version=%%I
 
 :final

